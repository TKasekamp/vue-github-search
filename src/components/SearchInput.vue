<template>
  <v-text-field
    v-model="message"
    append-outer-icon="mdi-send"
    filled
    clear-icon="mdi-close-circle"
    clearable
    label="Search repositories"
    type="text"
    :loading="loading"
    :readonly="loading"
    @click:append-outer="sendMessage"
    @click:clear="clearMessage"
 />
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'SearchInput',

  data: () => ({
    show: false,
    message: '',
    marker: true,
  }),

  props: {
    loading: Boolean,
  },

  methods: {
    sendMessage() {
      // this.clearMessage();
    },
    clearMessage() {
      this.message = '';
    },
  },
});
</script>
