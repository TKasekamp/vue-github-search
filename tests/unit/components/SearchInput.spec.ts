import { shallowMount, createLocalVue } from '@vue/test-utils';
import SearchInput from '@/components/SearchInput.vue';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(Vuetify);

describe('SearchInput.vue', () => {
  const createWrapper = (options: object) => shallowMount(SearchInput, {
    localVue,
    vuetify: new Vuetify(),
    ...options,
  });

  describe('when loading true', () => {
    it('is disabled and loading', () => {
      const wrapper = createWrapper({
        propsData: { loading: true },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
