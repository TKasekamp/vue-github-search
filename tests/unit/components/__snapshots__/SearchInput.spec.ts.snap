// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput.vue when loading true is disabled and loading 1`] = `<v-text-field-stub errorcount="1" errormessages="" messages="" readonly="true" rules="" successmessages="" value="" backgroundcolor="" label="Search repositories" loading="true" loaderheight="2" appendoutericon="mdi-send" clearable="true" clearicon="mdi-close-circle" filled="true" type="text"></v-text-field-stub>`;
